dataprofiler/profilers/profile_builder.py

```diff
@@ -2088,11 +2088,13 @@ def _update_row_statistics(
 
         # If sample_ids provided, increment since that means only new data read
         if sample_ids is not None:
-            self.row_has_null_count += len(null_in_row_count)
-            self.row_is_null_count += len(null_rows)
+            if self.options.row_statistics.null_count.is_enabled:
+                self.row_has_null_count += len(null_in_row_count)
+                self.row_is_null_count += len(null_rows)
         else:
-            self.row_has_null_count = len(null_in_row_count)
-            self.row_is_null_count = len(null_rows)
+            if self.options.row_statistics.null_count.is_enabled:
```

Comment:
what if we move this `if` outside the other `if`. Then we only have 1 conditional as oppose to 2